sympy/functions/elementary/piecewise.py

```diff
@@ -338,16 +377,14 @@ def _sort_expr_cond(self, sym, a, b, targetcond=None):
                         raise NotImplementedError(
                             "Unable to handle interval evaluation of expression.")
             else:
-                if sym not in [cond.lts, cond.gts]:
-                    cond = _solve_inequality(cond, sym)
                 lower, upper = cond.lts, cond.gts  # part 1: initialize with givens
                 if cond.lts == sym:                # part 1a: expand the side ...
-                    lower = S.NegativeInfinity   # e.g. x <= 0 ---> -oo <= 0
-                elif cond.gts == sym:            # part 1a: ... that can be expanded
-                    upper = S.Infinity           # e.g. x >= 0 --->  oo >= 0
+                    lower = S.NegativeInfinity     # e.g. x <= 0 ---> -oo <= 0
+                elif cond.gts == sym:              # part 1a: ... that can be expanded
+                    upper = S.Infinity             # e.g. x >= 0 --->  oo >= 0
                 else:
```

Comment:
I don't think this is necessary.

But the else could be changed to `elif isinstance(cond, Relational):` and an `else` clause added to gracefully handle the unexpected.